### PR TITLE
Some small changes to improve readability dealing with claim service results.

### PR DESCRIPTION
**Pull request by zheileman**
Also removing the coupling with Google Analytics, and returning the service result instead of the service once we run call()

### CODE REVIEW

**Comment by slorek** [comment]
Definitely better now. Not directly relevant to this PR but consider having the services' `call` method return the `result` object.

Also consider moving the instantiation of the `Claim` object (`@claim`) into the service in all cases.

File: app/controllers/external_users/claims_controller.rb [hunk 1]
```diff
@@ -344,15 +342,15 @@ def render_action_with_resources(action)
...
 
-  def handle_submit_result(service)
+  def render_or_redirect(service)
     if service.result.success?
-      handle_submit_success(service)
+      render_or_redirect_success(service)
     else
-      handle_submit_error(service)
+      render_or_redirect_error(service)
     end
```
Comment (slorek) [inline app/controllers/external_users/claims_controller.rb]: Consider the following:

``` ruby
return render_or_redirect_error(service) unless service.result.success?
render_or_redirect_success(service)
```

**Comment by slorek** [comment]
Made some minor changes. Still quite a bit to do, but will be done in another PR to prevent changes becoming stale.

**Comment by zheileman** [comment]
@slorek happy with the changes. Are we gonna merge this PR? If so, go ahead :thumbsup:

**Pull request merged**